### PR TITLE
optimize(hz): slice method for thrift

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### What type of PR is this?

optimize
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) Translate the PR title into Chinese.
优化 thrift IDL 声明接口路由的方式

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional): thrift 的接口定义可指定多个相同的 http method 定义，例如
` (api.get="/hello", api.get="hello2", api.get="hello3");`

### CODE REVIEW

File: cmd/hz/thrift/ast.go [hunk 1]
```diff
@@ -199,17 +199,30 @@ func astToService(ast *parser.Thrift, resolver *Resolver, args *config.Argument)
...
+						tmp.GenHandler = false
+						methods = append(methods, &tmp)
+					}
 					continue
 				}
-				tmp := *method
-				hmethod, path := anno.method, anno.path
-				if len(path) != 1 || path[0] == "" {
-					return nil, fmt.Errorf("invalid api.%s  for %s.%s: %s", hmethod, s.Name, m.Name, path)
+				for i := 0; i < len(anno.path); i++ {
```
Comment (welkeyever) [inline cmd/hz/thrift/ast.go:215]: 这几段完全相同的代码可以复用一下：应该把 i 的初始化提出来就能实现

File: cmd/hz/thrift/ast.go [hunk 2]
```diff
@@ -199,17 +199,30 @@ func astToService(ast *parser.Thrift, resolver *Resolver, args *config.Argument)
...
+					tmp := *method
+					hmethod, path := anno.method, anno.path
+					if path[i] == "" {
+						return nil, fmt.Errorf("invalid api.%s  for %s.%s: %s", hmethod, s.Name, m.Name, path[i])
```
Comment (welkeyever) [inline cmd/hz/thrift/ast.go:219]: %s 后面多了个一个空格？

Comment (FGYFFFF) [reply cmd/hz/thrift/ast.go]: done

File: cmd/hz/thrift/ast.go (hunk 1 above)
Comment (FGYFFFF) [reply cmd/hz/thrift/ast.go:201]: done

File: cmd/hz/thrift/ast.go [hunk 3]
```diff
@@ -198,18 +198,16 @@ func astToService(ast *parser.Thrift, resolver *Resolver, args *config.Argument)
...
 			for idx, anno := range httpAnnos {
-				if idx == 0 {
-					continue
-				}
-				tmp := *method
-				hmethod, path := anno.method, anno.path
-				if len(path) != 1 || path[0] == "" {
-					return nil, fmt.Errorf("invalid api.%s  for %s.%s: %s", hmethod, s.Name, m.Name, path)
+				for i := 0; i < len(anno.path); i++ {
+					if idx == 0 { // idx==0 && i==0 has been added above
```
Comment (welkeyever) [inline cmd/hz/thrift/ast.go:202]: 看着有点奇怪，202行为啥不放在201行上面？

Comment (FGYFFFF) [reply cmd/hz/thrift/ast.go]: 👍👍👍 这块漏了个判断条件，改完了

**Review by welkeyever — APPROVED** [review]
LGMT

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**